### ISSUE TITLE
Does not work in Vim8

### ISSUE THREAD

**Issue by ferguscollins** [spinks/vim-leader-guide#7]
The plugin calls references to the functions `nvim_create_buf` and `nvim_create_win` which do not seem to exist in Vim, causing the plugin to error out upon the first call of `leaderGuide#start_by_prefix`. I am assuming these are defined in NVim and that this is why it is incompatible. Is there an alternative that will accomplish the same goal in Vim, and are there other incompatibilities elsewhere?

**Comment by spinks** [comment]
Yes, when I swapped it over to NVim floating windows the Vim alternative was not as well defined and as such I just took on the NVim specific functions (in effect dropping regular vim support).

There is now better support for a floating style window through the Popup API, and so it would be possible to create window creation functions around that method instead of the NVim methods. Working with the popup windows and creating/writing buffers in vim is a bit less elegant iirc so would take some work to adapt.

I believe that the only incompatibilities would lie in these window and buffer creation functions.

As a reference, vim-which-key implements Vim popup windows as such: [vim-which-key/window.vim](https://github.com/liuchengxu/vim-which-key/blob/30c0810b012a7acdccc6b72f0e99c0388986844f/autoload/which_key/window.vim#L58) (also illustrating the Vim method to create and write the buffer).

Alternatively, until the vim popup methods are implemented, you could drop back to the legacy branch, which has the old style windows but is vim compatible.
